src/components/vr_cooperation/src/command/base_json_request.cc

```diff
@@ -123,41 +123,35 @@ void BaseJsonRequest::ParseMobileResultCode(
   result_code = GetHMIResultCode(mobile_result_code);
 }
 
-void BaseJsonRequest::PrepareRequestMessageForMobile(
-    vr_hmi_api::RPCName function_id, const std::string& message_params,
+void BaseJsonRequest::SendMessageToHMI(
+    const vr_hmi_api::ServiceMessage& message) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  parent_->SendMessageToHMI(message);
+}
+
+void BaseJsonRequest::SendMessageToMobile(
     application_manager::MessagePtr message) {
   LOG4CXX_AUTO_TRACE(logger_);
+
+  message = new application_manager::Message(protocol_handler::MessagePriority::kDefault);
```

Comment:
I can not understand why do you use input argument here and replace object in shared pointer?